spec/evalf_spec.rb

```diff
@@ -0,0 +1,29 @@
+describe 'SymEngine::evalf' do
+    context 'RealDouble values' do
+        subject { SymEngine::evalf(SymEngine::sin(SymEngine(2)), 53, true) }
+        it { is_expected.to be_a SymEngine::RealDouble }
+        its(:to_s) { is_expected.to eq "0.909297426825682" }
+    end
+    if SymEngine::HAVE_MPFR
+        context 'RealMPFR values' do
+            subject { SymEngine::evalf(SymEngine::PI * SymEngine(1963319607) - SymEngine(6167950454), 100, true) }
+            it { is_expected.to be_a SymEngine::RealMPFR }
+            its(:to_s) { is_expected.to eq "1.4973429143989597928099399837265e-10" }
+        end
+    end
+    context 'ComplexDouble values' do
+        subject { SymEngine::evalf(SymEngine::sin(1) * SymEngine::I, 53, false) }
+        it { is_expected.to  be_a SymEngine::ComplexDouble }
+        its(:to_s) { is_expected.to eq "0.0 + 0.841470984807897*I" }
+    end
+    context 'ComplexMPC values' do
+    
+    end
+    if SymEngine::HAVE_MPC
+        context 'ComplexMPC values' do
+            subject { SymEngine::evalf(SymEngine::sin(1) * SymEngine::I, 100, false) }
+            it { is_expected.to  be_a SymEngine::ComplexMPC }
+            its(:to_s) { is_expected.to eq "0.00000000000000000000000000000000 + 0.84147098480789650665250232163005*I" }
```

Comment:
So to not lose precision we should use BigDecimal and then compare. In 5 and 17, if the numbers are BigDecimals, can they be compared then? Like
```ruby
# 5
BigDecimal("0.909297426825682")
# 17 
Complex(0.0, BigDecimal("0.841470984807897"))
```
There will be a little problem in line 26 but 11 can be done with `BigDecimal`.
In 26, `BigDecimal("0.00000000000000000000000000000000")` reduces the precision to `0.0`. I couldn't find a way to retain it as it is.